src/middlewares/validation.middleware.ts

```diff
@@ -27,29 +27,48 @@ async function compileAjv(options: ValidationMiddlewareOptions) {
   const pathName = options.path;
   const path = paths[String(pathName)];
   if (!path) {
-    throw new Error(`Path "${pathName}" doesn't exist in the OpenAPI document.`);
+    throw new Error(
+      `Path "${pathName}" doesn't exist in the OpenAPI document.`
+    );
   }
-  
+
   const methodName = options.method;
   const method = path[String(methodName)];
   if (!method) {
-    throw new Error(`Method "${methodName}" for "${pathName}" path doesn't exist in the OpenAPI document.`);
+    throw new Error(
+      `Method "${methodName}" for "${pathName}" path doesn't exist in the OpenAPI document.`
+    );
   }
 
   const requestBody = method.requestBody;
-  if (!requestBody) return;
+  const requestParams = method.parameters;
+  if (!requestBody && !requestParams) return;
+  let schema;
+  if (requestBody) {
+    schema = requestBody.content['application/json'].schema;
+  }
+  if (requestParams) {
+    const newObj = {};
+    requestParams.map((param) => {
+      newObj[param.name] = param.schema;
+    });
+    schema = {
+      properties: newObj,
+    };
+  }
```

Comment:
You wrote logic to handle schema only for `requestBody` or `parameters`. Try to fix this. That function should return validate functions for `requestBody` and `parameters`.

Reply:
@magicmatatjahu but what if the request body isn't available in the request? like in the case of a get request

Reply:
You can have post with request body and parameters. It needs to be generic.